emulator/src/core/emulator.rs

```diff
@@ -394,6 +398,47 @@ impl Emulator {
     pub fn load_ram(&mut self, data: Vec<u8>, start: u16) {
         self.ram.load_vec(data, start)
     }
+
+    pub fn interrupt(&mut self, opcode: u8) -> EResult<()> {
```

Comment:
Maybe I am not aware enough of how interrupts work, but isn't this change just encapsulating the original `execute_next()` method? What's the reason behind this?

Reply:
An external device calls an interrupt by sending one opcode to the CPU which is executed (typically a RST instruction).

Reply:
But doesn't this method call do the same as the method we used earlier? I'm only wondering if this is basically a refactor to better match the naming.

Reply:
The method will also need to do some other stuff (see my change request)